numpy/core/src/multiarray/methods.c

```diff
@@ -371,6 +371,7 @@ PyArray_GetField(PyArrayObject *self, PyArray_Descr *typed, int offset)
         safe = PyObject_CallFunction(checkfunc, "OOi", PyArray_DESCR(self),
                                      typed, offset);
         if (safe == NULL) {
+            Py_DECREF(typed);
```

Comment:
Needed above on line 366 too.

Reply:
whoops, I think SetField I did not forget that, will add that right away.